tests/test_pydantic_utils.py

```diff
@@ -137,6 +142,24 @@ def test_time(self) -> None:
         with self.assertRaises(ValueError):
             adapter.validate_json({})
 
+    def test_ellipsis(self) -> None:
+        adapter = pydantic.TypeAdapter(SerializableEllipsis)
+        json_roundtripped = adapter.validate_json(adapter.dump_json(...))
+        self.assertIs(json_roundtripped, ...)
+        python_roundtripped = adapter.validate_python(adapter.dump_python(...))
+        self.assertIs(python_roundtripped, ...)
```

Comment:
Oh, bleh.

I tried rewriting this with `__get_pydantic_core_schema__` (since I understand that better), but any solution that fixes this problem seems to fail to work with the union.  I'm inclined to go replace ellipsis usage here with a single-variant `enum`; I had that idea later in the ticket and wondered if it might be better but didn't see a need, but I think this problem puts me on the other side of that fence.

Reply:
Yeah the enum thing will probably work, or one thing that I often see in JSON is using an object like `{ "special": "any_type" }`, which can't be confused with a string.  (Better if you know you need it up front so you end up with a discriminated union instead of an awkward union of string and object, but `str | object` ends up happening a lot.)

Reply:
Redone with an enum.  Turns out it's important to set the enum value to something that a string (I used `-1`) since that's what Pydantic uses for serialization, and it's not able to handle unions whose members are overlapping in the JSON space.